### ISSUE TITLE
Rate limiting being triggered on web assets

### ISSUE THREAD

**Issue by BrokenEagle** [r888888888/danbooru#4090]
> albert said ([forum #1234](https://danbooru.donmai.us/forum_posts/157243)):
> I've bumped the limit on posts to 10 requests/second (up from 4) and added one for users to 5 request/second.

I just noticed a problem with the above. It now triggers on the **custom_style.css** file.

> users/custom_style.css:1 Failed to load resource: the server responded with a status of 429 ()

The above happened with the batch bookmarklet "Open all links in new windows" with 4 image links. The rate limiting shouldn't trigger on web assets like that, as those are often important to the functionality of the page being returned.

I'm not sure how up-to-date the custom style needs to be anyways. If hammering the server is a problem, perhaps the custom style could be cached for a minute.  Or would there be a way to track changes to the custom style and somehow use that to control caching?

**Comment by kittey** [comment]
A common technique is to include the last modification time as a parameter to what you’re caching, then setting the cache lifetime to infinite. Suppose I modified my custom CSS yesterday, the site would request custom_style.css?1560213752. The parameter would be ignored and the content would be returned with infinite cache lifetime. If I then modified my custom CSS just now, the site would then request custom_style.css?1560300125. The different parameter would make any cache request the data from the server again.

If you’re storing the user’s custom CSS last modification time, it would be possible to store a modification time of 0 for empty custom CSS and then not request the custom CSS stylesheet at all.
